pkg/wireguard/agent/agent.go

```diff
@@ -109,6 +109,11 @@ func (a *Agent) Init() error {
 	link := &netlink.Wireguard{LinkAttrs: netlink.LinkAttrs{Name: types.IfaceName}}
 	err := netlink.LinkAdd(link)
 	if err != nil && !errors.Is(err, unix.EEXIST) {
+		if errors.Is(err, unix.EOPNOTSUPP) {
+			return fmt.Errorf("wireguard not supported by the Linux kernel (netlink: %w). "+
+				"Please ensure your system has the wireguard kernel module installed: "+
```

Comment:
Nitpicking, but this might be misleading to users, as they might think that Wireguard is an out-of-tree module. Maybe give a hint about kernel version?

Reply:
I think providing a kernel version is not a good idea. Some distributions like Ubuntu 20.04 have backported Wireguard to older kernels. Other distributions like Debian or RHEL actually do provide the out-of-tree module which is compatible with the `wgctrl` client.

Reply:
Then maybe `upgrade kernel or install manually`?

Reply:
Changed to `Please upgrade your kernel or manually install the kernel module`